src/fluxpy/utils/model.py

```diff
@@ -415,3 +419,309 @@ def sync_with_medium(model: cobra.Model, medium: Dict):
+
+            
+            # if 2 or more reactions match the minimum number of cofactors
+            if len(keep_indices_group) > 1:
+                # find which reaction has the most abundant cofactor (general abundance from the model) 
+                # if the most abundant cofactor appears in all reactions, then check the next most abundant cofactor
+                
+                
+                def order_cofactors_by_abundance(sorted_counts_data, specific_reaction_cofactors):
+                    
+                    """
+                    Function that takes the cofactors of a single reaction and sorts them 
+                    based on their overall abundance in the model
+
+                    Example input:
+                    
+                    sorted_counts_data = [('h_c', 35), ('h2o_c', 18), ('h_e', 17), ('atp_c', 13), 
+                                        ('adp_c', 12), ('nad_c', 12), ('nadh_c', 12), ('pi_c', 12)]
+                                        
+                    specific_reaction_cofactors = ['h_e', 'h_c', 'adp_c', 'nadh_c']
+                    
+                    Example output:
+                    
+                    [35, 17, 12, 12]
+                    """
+
+                    # Convert the list of tuples to a dictionary for fast lookup
+                    abundance_dict = dict(sorted_counts_data)
+                    
+                    # Sort the items based on their abundance in the dictionary
+                    sorted_cofactors = sorted(specific_reaction_cofactors, key=lambda x: abundance_dict.get(x, 0), reverse=True)
+                    sorted_abundances = [abundance_dict.get(item, 0) for item in sorted_cofactors]
+                    
+                    return sorted_abundances
+                     
+                      
+                def find_first_winning_sublist(sublists):
+                    
+                    """
+                    Function to find which reaction to keep from a certain group 
+
+                    Example input:
+                    
+                    sublists = [
+                        [9, 3, 5, 7],
+                        [9, 3, 4, 6],
+                        [0, 5, 3, 3],
+                        [4, 2, 1, 6],
+                        [7, 3, 2, 0]
+                    ]
+                    
+                    Example output:
+                    
+                    [9, 3, 5, 7]
+                    
+                    """
+                    
+                    num_sublists = len(sublists)
+                    max_index = 0  # Start with the first sublist as the default winner
+                    losers_indices = []  # List to store indices of non-winning sublists
+                    
+                    # Loop through the sublists to compare each one
+                    for i in range(1, num_sublists):
+                        for j, (a, b) in enumerate(zip(sublists[max_index], sublists[i])):
```

Comment:
Why do you use `enumerate`? The `j` variable is not being used.
Also, are you sure there is no easier way to find the first winning sublist? 
Is this function really needed? 

Could you not go with something like:

```python
def find_first_winning_sublist_with_losers(sublists):
    """
    Find the "winning" sublist and indices of non-winning sublists.

    Parameters:
        sublists (list of lists): Input list of sublists to compare.

    Returns:
        tuple: (winning_sublist, losers_indices)
    """
    winning_sublist = max(sublists, key=lambda x: x)
    losers_indices = [i for i, sublist in enumerate(sublists) if sublist != winning_sublist]
    return winning_sublist, losers_indices
```